src/modules/VirtConfig.rb

```diff
@@ -317,6 +318,70 @@ def LXCDialog
         )
       end
 
+      Builtins.y2milestone("VirtConfig::ConfigureDom0: Checking for Installed Patterns and Packages")
+      is_xen_server_installed = false
+      is_xen_tools_installed = false
+      is_kvm_server_installed = false
+      is_kvm_tools_installed = false
+      is_client_tools_installed = false
+      if isOpenSuse == true
+        # On openSUSE there are no 'tools' patterns for Xen and KVM
+        if Package.Installed("patterns-openSUSE-xen_server")
+          is_xen_server_installed = true
+        end
+        if Package.Installed("xen-tools") && Package.Installed("xen-libs") &&
+           Package.Installed("libvirt-daemon-xen") && Package.Installed("tigervnc") &&
+           Package.Installed("virt-manager")
+          is_xen_tools_installed = true
+        end
+        if Package.Installed("patterns-openSUSE-kvm_server")
+          is_kvm_server_installed = true
+        end
+        if Package.Installed("libvirt-daemon-qemu") && Package.Installed("tigervnc") &&
+           Package.Installed("virt-manager")
+          is_kvm_tools_installed = true
+        end
+      else
+        if isSLED == true
+          # On SLED there is only a 'virtualization_client' pattern
+          if Package.Installed("patterns-sled-virtualization_client")
+            is_client_tools_installed = true
+          end
+        else
+          if Package.Installed("patterns-sles-xen_server")
+            is_xen_server_installed = true
+          end
+          if Package.Installed("patterns-sles-xen_tools")
+            is_xen_tools_installed = true
+          end
+          if Package.Installed("patterns-sles-kvm_server")
+            is_kvm_server_installed = true
+          end
+          if Package.Installed("patterns-sles-kvm_tools")
+            is_kvm_tools_installed = true
+          end
+        end
+      end
+      # Desensitize CheckBox if pattern or packages are already installed
+      if is_xen_server_installed
```

Comment:
Enabling already enabled widgets is cheap, I'd prefer single line `UI.ChangeWidget(Id(:xen_server), :Enabled, !is_xen_server_installed)` for disabling the check boxes.